File: Yolların sayı.py
# https://basecamp.eolymp.com/az/problems/11114
MOD = 998244353

def power(n, m):
    if m <= 1:
        return n % MOD
    elif m % 2 == 0:
        return power((n % MOD) * (n % MOD), m // 2) % MOD
    else:
        return (n % MOD) * power(n, m - 1) % MOD

def main():
    n, m = map(int, input().split())
    res = power(2 * n + 1, m) % MOD
    print(res)

if __name__ == "__main__":
    main()

File: Yolların sayı.cpp
// https://basecamp.eolymp.com/az/problems/11114
#include<stdio.h>
#include<math.h>
#define ull unsigned long long
const int MOD=998244353;
ull power(ull n, int m){
	if(m<=1){
	return n%MOD;
	}else if(m%2==0){
		return power((n%MOD)*(n%MOD),m/2)%MOD;
	}else{
		return n%MOD*power(n,m-1);
	}
}
int main(){
	int n,m;
	scanf("%d%d",&n,&m);
	ull res=power(2*n+1,m)%MOD;
	printf("%lld",res);
}